test/common/chunked_array_test.cpp

```diff
@@ -0,0 +1,64 @@
+#include "common/container/chunked_array.h"
+
+#include <cstring>
+#include <random>
+#include <thread>  // NOLINT
+#include <unordered_set>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+namespace noisepage {
+
+template <int chunk>
+void FillMergeSize(size_t size1, size_t size2) {
+  std::default_random_engine generator;
+  auto dist = std::uniform_int_distribution(1, 1000);
+
+  common::ChunkedArray<int, chunk> array1;
+  common::ChunkedArray<int, chunk> array2;
+  std::vector<int> vec;
+  for (size_t i = 0; i < size1; i++) {
+    int num = dist(generator);
+    vec.push_back(num);
+    array1.Push(std::move(num));  // NOLINT
+  }
+
+  for (size_t i = 0; i < size2; i++) {
+    int num = dist(generator);
+    vec.push_back(num);
+    array2.Push(std::move(num));  // NOLINT
+  }
+
+  array1.Merge(&array2);
+
+  size_t idx = 0;
+  auto it = array1.begin();
+  while (it != array1.end()) {
+    ASSERT_EQ(vec[idx], *it);
+    it++;
+    idx++;
+  }
+
+  ASSERT_EQ(idx, size1 + size2);
```

Comment:
We should use `EXPECT_EQ` etc. in tests instead of the asserts. Then it would integrate nicely with gtest. The same applies to the query trace logging test.